src/main/java/graphql/nadel/NadelExecutionHints.java

```diff
@@ -0,0 +1,33 @@
+package graphql.nadel;
+
+public class NadelExecutionHints {
+    private final boolean optimizeExecution;
+
+    private NadelExecutionHints(boolean optimizeExecution) {
+        this.optimizeExecution = optimizeExecution;
+    }
+
+    public static Builder newHints() {
+        return new Builder();
+    }
+
+    public boolean getOptimizeExecutionFlag() {
+        return optimizeExecution;
+    }
+
+    public static class Builder {
+        private boolean optimizeExecution;
+
+        private Builder() {
+        }
+
+        public Builder optimizeExecution(boolean flag) {
+            this.optimizeExecution = flag;
+            return this;
+        }
+
+        public NadelExecutionHints build() {
+            return new NadelExecutionHints(optimizeExecution);
```

Comment:
I know we only have 1 flag for now but I think this should be generalised so it returns NadelExecutionHints(this) rather than a single flag.